sway/ipc-json.c

```diff
@@ -268,6 +268,28 @@ static void ipc_json_describe_view(struct sway_container *c, json_object *object
 				json_object_new_int(view_get_x11_window_id(c->view)));
 	}
 #endif
+
+	/*
+	 * The "window_properties" object is mostly redundant with other existing
+	 * fields in the view object, but is still somewhat necessary for drop-in
+	 * compatibility with existing i3 ipc tools/libs that expect them
+	 * (e.g. the i3ipc python module).
+	 */
+	json_object *window_props = json_object_new_object();
+
+	// Some tools for i3 break when windows don't have a class.
+	if (!class) {
+		class = app_id;
```

Comment:
-1 to this. Loading the app id into the class or instance fields makes it confusing to end users. I value transparency - that is: telling the user what the properties actually are - over trying to allow for a zero configuration port between i3 and sway. I add an `app_id` property, set it accordingly, and let the class and instance be null. This is the approach we've taken with criteria too.

Reply:
Wow, sorry for the spam, looks like [GitHub is choking on requests right now](https://blog.github.com/2018-10-21-october21-incident-report/).

Reply:
I don't have a problem with setting I3SOCK because there's no bad consequences for doing it. But putting the app id into the class and instance is a decision that we could be forced to live with for a while, can confuse and frustrate users, and will become a legacy (especially when users no longer run Xwayland).

Reply:
As a compromise, maybe set the class to _wayland_ if it is a native wayland window?

Reply:
Alright. I somewhat find it just as frustrating to see every single ipc script that check the instance or the class of a window completely break, but I can see why you wouldn't want that in sway. I just hoped that sway would own being its own thing for the sake of consistency.

I'll just write a standalone ipc proxy providing an I3SOCK socket to convert from the sway-flavored json to the i3-flavored one in the meantime.

tl;dr: Dropping this PR; window_properties are very much more so a compatibility thing with i3 rather than something critical.